src/array/map/index.ts

```diff
@@ -0,0 +1,6 @@
+/**
+ * Applies the given function to every element of an array.
+ *
+ * @param fn  A function to map an element
+ */
+export const map = <A, B>(fn: (value: A) => B) => (values: A[]) => values.map(fn)
```

Comment:
Hey @raxell, thanks for the Pull Request 🙏

Would you mind updating the `fn` signature to include the other map parameters?
Using [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) as a reference, since the second and third parameters are optional I think this function should be overloaded.